raiden_contracts/data/source/raiden/TokenNetwork.sol

```diff
@@ -789,10 +791,10 @@ contract TokenNetwork is Utils {
             participant2_settlement
         );
 
-        // Remove the channel data from storage
-        delete channel.participants[participant1];
-        delete channel.participants[participant2];
-        delete channels[channel_identifier];
+        // Update channel state
+        participant1_state.withdrawn_amount += participant1_settlement.transferred_amount;
+        participant2_state.withdrawn_amount += participant2_settlement.transferred_amount;
+        channel.state = ChannelState.NonExistent;
```

Comment:
That feels a bit weird. But we don't get any advantage from differentiating more here.